polkadot/xcm/xcm-executor/src/lib.rs

```diff
@@ -866,8 +971,10 @@ impl<Config: config::Config> XcmExecutor<Config> {
 					let assets = Self::reanchored(deposited, &dest, None);
 					let mut message = vec![ReserveAssetDeposited(assets), ClearOrigin];
 					message.extend(xcm.0.into_iter());
-					// put back transport_fee in holding register to be charged by XcmSender
-					self.holding.subsume_assets(transport_fee);
+					// put back delivery_fee in holding register to be charged by XcmSender
```

Comment:
it is important that there is no possibility to use a pool between quote and actual exchange, otherwise the price might change. if this is true now, I think it still better to try to keep those operations in the single context, otherwise it easy to break that requirements in the future.

Reply:
It's not going to be possible to use the pool between the quote and exchange since it's all part of the same instruction executions.
Users might be able to put their own `ExchangeAsset` instruction in but that'll happen afterwards or before, not in between.